### PR TITLE
(Fix)Deprecate PrettyState in python API

### PR DESCRIPTION
**Pull request by jona42-ui**
Issue: https://github.com/nmstate/nmstate/issues/2588

Description;

The PrettyState is only used for test codes.
    

- [x] Mark PrettyState as deprecated in Python API, providing message indicate libnmstate.show() output is already sorted.
- [x] Remove test cases regarding PrettyState

### CODE REVIEW

File: rust/src/python/libnmstate/prettystate.py
Comment (ffmancera) [inline rust/src/python/libnmstate/prettystate.py]: Instead of introducing the warning here, could you place it under the `__init__()` function in `PrettyState` class definition? This way the warning will be displayed when creating an instance of the class.

File: tests/integration/linux_bridge_test.py
Comment (ffmancera) [inline tests/integration/linux_bridge_test.py]: The whole test case needs to be droppped. The test is specific to pretty_state that we do not want to test anymore. Please drop the whole `test_pretty_state_port_name_first` test.

**Review by ffmancera — CHANGES_REQUESTED** [review]
There is also a missing place where `PrettyState` needs to be dropped. `tests/integration/ovs_test.py`

**Comment by ffmancera** [comment]
Hi! Instead of creating multiple commits could you unify all the changes into a single one? In addition, instead of doing merges to update the branch, please do rebases. The git history will be more clear

**Comment by jona42-ui** [comment]
cleaning this up, sorry!

**Comment by ffmancera** [comment]
It looks much better but the commit message it reflects messages from other commits. I guess this is because you squashed the commits. Please, adjust the commit message to reflect only the relevant information of this task :)

**Comment by jona42-ui** [comment]
updated, thanks

**Review by ffmancera — APPROVED** [review]

**Comment by ffmancera** [comment]
LGTM! Let's wait for CI results before merging it

**Comment by ffmancera** [comment]
Whoops, the linter is failing. Could you follow the formatting suggestion? https://github.com/nmstate/nmstate/actions/runs/8202211120/job/22432495995?pr=2591#step:3:436

```
 class PrettyState:
     def __init__(self, state):
         yaml.add_representer(dict, represent_dict)
-        warnings.warn("The PrettyState class is deprecated and will be removed in a future release. "
-                      "The output of libnmstate.show() is already sorted.", DeprecationWarning)
+        warnings.warn(
+            "The PrettyState class is deprecated and will be removed in a future release. "
+            "The output of libnmstate.show() is already sorted.",
+            DeprecationWarning,
+        )
         self.state = _sort_with_priority(state)
```

**Comment by jona42-ui** [comment]
the linter is quite annoying!

**Comment by ffmancera** [comment]
> the linter is quite annoying!

It might be but it is important to take care of the style used when coding. An unified and clean style leads to a codebase easier to read and understand.

There is one more lint error, you are almost there!

```
 +
 class PrettyState:
     def __init__(self, state):
         yaml.add_representer(dict, represent_dict)
         warnings.warn(
             "The PrettyState class is deprecated and will be removed in a future release. "
```

**Comment by ffmancera** [comment]
All good, merging!

**Comment by cathay4t** [comment]
According to https://peps.python.org/pep-0702/  we should use `@warnings.deprecated()`. 

@jona42-ui Can you investigate and fix it in next PR?

**Comment by jona42-ui** [comment]
thanks makes sense

**Comment by jona42-ui** [comment]
can i make a followup PR with only the decorator fix?

**Pull request merged**
